Topics/Tech_Stacks/CSS-FlexBox.md

```diff
@@ -0,0 +1,124 @@
+-	Inline: To organize texts in our webpage
+-	Table: To organize in a 2D-grid based system
+-	Block: To organize grouped or sections of our webpage
+-	Positioned: To specify exact position for items. 
+
+These four layout modes required lots of code to do some simple tasks like “Centering a div”. This is where FlexBox comes in clutch! This not only made coding easier, but also opened CSS Webpage-organization to a whole new level. 
+
+<img width="541" alt="flexbox" src="https://github.com/Raiyankr/CSS-Flexbox/assets/110127056/ebbf9c61-a5db-4f62-bbb7-010acd17009d">
+
+[Image source](https://acciojob.com/blog/untitled-19/)
+
+## How to use FlexBox?
+
+There are two things we need to understand in order to use FlexBox. 
+- Flex Container: This container will contain some items from our webpage. This container will also define the organization property of the items combined.
+- Flex Item: These can be individual components in our webpage (anywhere from a div, to a nested div).
+
+Before we learn about flex containers, lets talk about flex items. These are any component that is nested or "contained" in our flex container. 
+
+```
+<div class="flex-container">
+  <div><1/div>
+  <div>2</div>
+  <div>3</div>
+  <div>4</div>
+</div>
+```
+![flex-item-example](https://github.com/Raiyankr/CSS-Flexbox/assets/110127056/31a39299-5dc9-42a2-b503-c1b9f038012d)
+[Image source](https://www.w3schools.com/css/css3_flexbox_container.asp)
+
+In the above pictures, we have four items in our flex-container. These items will appear in the same order we add them in. It is sufficient to know this much about flex items for now. Let us now learn how we can create a Flex Container. We can do so by using  "_Display: Flex_" property. This simply makes our container *Flex 💪🏻*. 
+
+```
+.flex-container{
+  display: flex;
+}
+```
+
+There are lots of other properties that we can play around with in our container such as:
+- flex-direction
+- flex-wrap
+- justify-content
+- align-items
+
+These properties can be used to change how we want to oprganize our items (vertically or horizontally), whether we want to center our whole container, whether we want to center our items, how do we want to space the items, etc. The following image shows how we can play around with the flex propoerties to do various tasks:
+
+![flex-properties](https://github.com/Raiyankr/CSS-Flexbox/assets/110127056/27c9a05e-7c0d-472a-b3f7-8afc27a80373)
+[Image source](https://www.bitdegree.org/learn/css-flexbox)
+
+To learn more about these properties, [W3Schoole](https://www.w3schools.com/css/css3_flexbox.asp) and [BitDegree](https://www.bitdegree.org/learn/css-flexbox) are great resources!
+
+## Tips and Tricks 
+
+One of the most commonly used property is "_flex-direction_". This property allows us to change the default axis from horizontal to vertical. We can see an example of this below:
+
```

Comment:
Hmm I included all of the tips under the same section thinking it would be appropriate to have them that way without being confusing to the reader. I will separate them to make it more clear, thanks for pointing it out.

Reply:
I separated them into three sub-sections now. Hope this makes it more clear, let me know if there is still any confusion!

Reply:
Looks good now! I was just trying to point out how each paragraph of text wasn't clear which property it was talking about at first glance, but it's good now